From f4baaac4eb396636cbb4e64ae7f82a683bb2b9c8 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Wed, 14 Mar 2018 10:45:10 +0800
Subject: [PATCH 3483/5242] MLK-17796: hdp: Remove duplicate define variable
 cur_mode

commit  da0fdc841003bf0242dc3fe7a461f9db423d8be9 from
https://source.codeaurora.org/external/imx/linux-imx.git

variable cur_mode have define in struct of imx_hdp.video.
so remove it in stuct imx_hdp.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp-audio.c |    2 +-
 drivers/gpu/drm/imx/hdp/imx-hdp.c       |    1 -
 drivers/gpu/drm/imx/hdp/imx-hdp.h       |    1 -
 3 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp-audio.c b/drivers/gpu/drm/imx/hdp/imx-hdp-audio.c
index 6624b8a..05cfe89 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp-audio.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp-audio.c
@@ -66,7 +66,7 @@ static u32 imx_hdp_audio(struct imx_hdp *hdmi, AUDIO_TYPE type, u32 sample_rate,
 	AUDIO_WIDTH bits;
 	int ncts_n;
 	state_struct *state = &hdmi->state;
-	int idx_n = select_N_index(hdmi->cur_mode->clock);
+	int idx_n = select_N_index(hdmi->video.cur_mode.clock);
 
 	switch (sample_rate) {
 	case 32000:
diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index 59dd43b..cb66689 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -483,7 +483,6 @@ static void imx_hdp_mode_setup(struct imx_hdp *hdp, struct drm_display_mode *mod
 
 	/* Get vic of CEA-861 */
 	hdp->vic = drm_match_cea_mode(mode);
-	hdp->cur_mode = mode;
 }
 
 bool imx_hdp_bridge_mode_fixup(struct drm_bridge *bridge,
diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.h b/drivers/gpu/drm/imx/hdp/imx-hdp.h
index 61759d4..472335b 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.h
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.h
@@ -232,7 +232,6 @@ struct imx_hdp {
 	VIC_PXL_ENCODING_FORMAT format;
 	bool hdr_metadata_present;
 	bool hdr_mode;
-	struct drm_display_mode *cur_mode;
 };
 
 void imx_hdp_register_audio_driver(struct device *dev);
-- 
1.7.9.5

